src/puppeteer-core.ts

```diff
@@ -195,21 +195,38 @@ class PuppeteerWorkers extends Puppeteer {
    * @param sessionId - sessionId obtained from a .sessions() call
    * @returns a browser instance
    */
-  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-  // @ts-ignore
   public override async connect(
     endpoint: BrowserWorker,
     sessionId: string
+  ): Promise<Browser>;
+
+  /**
+   * Establish a devtools connection to an existing session
+   *
+   * @param options - ConnectOptions
+   * @returns a browser instance
+   */
+  public override async connect(options: ConnectOptions): Promise<Browser>;
+
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  public override async connect(
+    endpointOrOptions: BrowserWorker | ConnectOptions,
+    sessionId?: string
   ): Promise<Browser> {
+    if (!sessionId) {
+      return super.connect(endpointOrOptions as ConnectOptions);
+    }
+    const endpoint = endpointOrOptions as BrowserWorker;
     try {
       const transport = await WorkersWebSocketTransport.create(
         endpoint,
-        sessionId
+        sessionId!
```

Comment:
Is the `!` necessary here? The check on line 217 should satisfy the undefined case. Below that point, the type of sessionId should just be `string`, not `string|undefined`